File: src/main/scala/sql/loadsave/SaveModeTest.scala
package main.scala.sql.loadsave

import org.apache.spark.sql.SQLContext
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.sql.SaveMode

object SaveModeTest {
  def main(args: Array[String]): Unit = {
     val conf = new SparkConf()
        .setAppName("GenericLoadSave")
        .setMaster("local")
    val sc = new SparkContext(conf)
    val sqlContext = new SQLContext(sc)
  
    //读取一个parquet文件
     /**
      * 将普通的文本格式转成parquet格式
      * 
      * val lineRDD = sc.textFile()
      * 将lineRDD变成DF
      * DF.save(parquet)
      */
     
    val usersDF = sqlContext.read.format("parquet").load("hdfs://hadoop1:9000/input/users.parquet")
    usersDF.save("hdfs://hadoop1:9000/output/users1.json", "json", SaveMode.Overwrite)
    
  }
}

File: src/main/java/sql/demo/SaveModeTest.java
package main.java.sql.demo;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.sql.DataFrame;
import org.apache.spark.sql.SQLContext;
import org.apache.spark.sql.SaveMode;

/**
 * 读取json或者parquet文件创建一个DataFrame
 * DataFrame存储到某一个路径下，默认存储格式是parquet
 * SaveMode.Overwrite：重写
 * Append:
 * @author root
 * 
 */

public class SaveModeTest {

	public static void main(String[] args) {
		SparkConf conf = new SparkConf()   
				.setAppName("SaveModeTest")
				.setMaster("local");
		JavaSparkContext sc = new JavaSparkContext(conf);
		SQLContext sqlContext = new SQLContext(sc);
	
		DataFrame peopleDF = sqlContext.read().json("hdfs://hadoop1:9000/input/people.json"); 
		
 		peopleDF.write().format("parquet").mode(SaveMode.Overwrite).save("hdfs://hadoop1:9000/output/0306parquet"); 
		
//		DataFrame df = sqlContext.read().parquet("hdfs://hadoop1:9000/output/0306parquet");
//		df.show();
	}
}





